assets/src/components/map.tsx

```diff
@@ -217,41 +223,108 @@ export const strokeOptions = ({ color }: Shape): object =>
         weight: 6,
       }
 
-const Shape = ({ shape }: { shape: Shape }) => {
-  const positions: LatLngExpression[] = shape.points.map((point) => [
-    point.lat,
-    point.lon,
-  ])
+const StopMarker = React.memo(
+  ({
+    stop,
+    iconSize,
+  }: {
+    stop: Stop
+    iconSize: StationIconSize
+  }): JSX.Element => {
+    if (stop.locationType === LocationType.Station) {
+      return <StationMarker station={stop} iconSize={iconSize} />
+    }
```

Comment:
**suggestion:** I think it might be a little easier to follow the logic if there were an umbrella component (`StopOrStationMarker`?) with logic that decides whether to render `StopMarker` or `StationMarker` rather than having that logic here.